window.py

```diff
@@ -9,62 +9,66 @@
+                directory = gui.popup_get_file('Select path', initial_folder=os.getcwd(), file_types=(("Executable files", "*.exe"),))
+                if directory == "":
+                    print("Nothing selected")
+                    continue
+                clean_dir = directory[(directory.rfind("/")+1):]
+                self.window["curExe"].update(value=clean_dir)
+                self.unpack_samase = directory
+            if event == "Launch":
+                if not pathSet(self.local_samase):
+                    continue
+                samase.run(self.local_samase, values["x64"])
+            if event == "Build From":
+                if not pathSet(self.local_samase):
+                    continue
+                samase.build(self.local_samase, values["x64"])
+            if event == "Unpack from":
+                if not pathSet(self.unpack_samase):
+                    continue
+                # on assumption that the user must select .exe file in the popup
+                assert self.unpack_samase.endswith(".exe"), f"Expected executable to unpack, instead got {self.local_samase}"
+                samase.build(self.unpack_samase, values["x64"])
 
-window = gui.Window("Sezam", layout, resizable=True)
-while True:
-    event, values = window.read()
-    if event == gui.WIN_CLOSED:
-        break
-    if event == "Select build directory":
-        directory = gui.popup_get_folder('Select path', initial_folder=os.getcwd())
-        if directory == "":
-            print("Nothing selected")
-            continue
-        clean_dir = directory[(directory.rfind("/")+1):] # TODO: use appropriate function from os.path
-        window["curPath"].update(value=clean_dir)
-        local_samase = directory
-    if event == "Select Executable to Unpack":
-        directory = gui.popup_get_file('Select path', initial_folder=os.getcwd(), file_types=(("Executable files", "*.exe"),))
-        if directory == "":
-            print("Nothing selected")
-            continue
-        clean_dir = directory[(directory.rfind("/")+1):]
-        window["curExe"].update(value=clean_dir)
-        unpack_samase = directory
-    if event == "Launch":
-        if not pathSet(local_samase):
-            continue
-        samase.run(local_samase, values["x64"])
-    if event == "Build From":
-        if not pathSet(local_samase):
-            continue
-        samase.build(local_samase, values["x64"])
-    if event == "Unpack from":
-        if not pathSet(unpack_samase):
-            continue
-        # on assumption that the user must select .exe file in the popup
-        assert unpack_samase.endswith(".exe"), f"Expected executable to unpack, instead got {local_samase}"
-        samase.build(unpack_samase, values["x64"])
+if __name__ == "__main__":
+    logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s', filename=LOGFILE, encoding='utf8', level = logging.INFO, datefmt='%Y-%m-%d %H:%M:%S')
+    logging.FileHandler(LOGFILE, mode = "w")
```

Comment:
Setting up logging seems like it's own thing that comes with just starting the program, and not really related to the window, which is why I think it makes sense to put it here.